libsolidity/codegen/MultiUseYulFunctionCollector.cpp

```diff
@@ -54,3 +70,15 @@ string MultiUseYulFunctionCollector::createFunction(string const& _name, functio
 	}
 	return _name;
 }
+
+string MultiUseYulFunctionCollector::createFunction(YulUtilFunction const& _function)
+{
+	solAssert(!_function.name.empty(), "");
+	solAssert(!_function.body.empty(), "");
+	if (!m_requestedFunctions.count(_function.name))
+	{
+		m_requestedFunctions[_function.name] = "<<STUB<<";
```

Comment:
Actually, I think that instead of accepting `YulUtilFunction` directly this should accept a `_creator` callback instead and the callback should be the one returning `YulUtilFunction`.

The other overload works like this to avoid deeply nested calls in code generator because one function might trigger generation of another and that one generation of another and so on. I'm not sure if the generator can actually handle generating multiple functions at the some time. `IRGeneratorContext` has stuff like `m_localVariables` and it might not be saved and then restored if you start generating a new function before you finish the previous one.